Regular Expressions/RegExp Common.sublime-syntax

```diff
@@ -0,0 +1,275 @@
+      push: inside-charset-range-begin
+    - include: character-class
+    - include: special-escaped-char
+    - include: escaped-char
+
+  inside-charset-range-begin:
+    - meta_content_scope: constant.other.range.regexp
+    - include: special-escaped-char
+    - include: escaped-char
+    - match: (?=-)
+      set: inside-charset-range-end
+
+  inside-charset-range-end:
+    - meta_scope: constant.other.range.regexp
+    - match: '-'
+      scope: punctuation.separator.sequence.regexp-common
+    - match: '{{known_char_escape}}'
+      scope: constant.character.escape.regexp
+      pop: true
+    - match: '{{invalid_char_escape}}'
+      scope: invalid.illegal.character.escape.regexp
+      pop: true
+    - match: '{{char_escape}}'
+      scope: constant.character.escape.regexp
+      pop: true
+    - match: .
+      pop: true
+
+  character-class:
+    - match: '{{char_class}}'
+      scope: keyword.control.character-class.regexp
+
+  special-escaped-char:
+    - match: '{{known_char_escape}}'
+      scope: constant.character.escape.regexp
+    - match: '{{invalid_char_escape}}'
+      scope: invalid.illegal.character.escape.regexp
+
+  escaped-char:
+    - match: '{{char_escape}}'
+      scope: constant.character.escape.regexp
+
+  backslash-q-literal:
+    - match: \\Q
+      scope: keyword.control.regexp
+      push: literal-until-backslash-e
+
+  literal-until-backslash-e:
+    - meta_content_scope: meta.literal.regexp
+    - match: \\E
+      scope: keyword.control.regexp
+      pop: true
+
+  backslashes:
+    - match: \\[bBAZzG]|[\^$]
+      scope: keyword.control.anchors.regexp
+      push: unexpected-quantifier-pop
+    - include: backslash-q-literal
+    - match: \\K
+      scope: keyword.control.regexp
+      push: unexpected-quantifier-pop
+    - match: \\([1-9]\d*)
+      scope: keyword.other.backref-and-recursion.regexp
+      captures:
+        1: variable.other.backref-and-recursion.regexp
+
+  quantifiers:
+    - match: '{{ranged_quantifier}}{{lazy_or_possessive}}'
+      scope: keyword.operator.quantifier.regexp
+      push: unexpected-quantifier-pop
+    - match: '{{character_quantifier}}{{lazy_or_possessive}}'
+      scope: keyword.operator.quantifier.regexp
+      push: unexpected-quantifier-pop
+
+  unexpected-quantifier:
+    - match: '{{ranged_quantifier}}{{lazy_or_possessive}}'
+      scope: invalid.illegal.unexpected-quantifier.regexp
+    - match: '{{character_quantifier}}{{lazy_or_possessive}}'
+      scope: invalid.illegal.unexpected-quantifier.regexp
+
+  unexpected-quantifier-pop:
+    - include: unexpected-quantifier
+    - match: ''
+      pop: true
+
+  operators:
+    - match: \|
+      scope: keyword.operator.alternation.regexp
+      push: unexpected-quantifier-pop
+
+  literal:
+    - include: quantifiers
+    - match: \.
+      scope: keyword.other.any.regexp # https://github.com/sublimehq/Packages/issues/314
+    - match: \)
+      scope: invalid.illegal.unmatched-brace.regexp
+    - match: \w+|.
```

Comment:
I'd avoid matching arbritary single characters like `|.` does, because the creates a token for every single - in this case non-word - character then.

See:

![grafik](https://user-images.githubusercontent.com/16542113/132948266-26ca2e61-e89b-4203-8772-94bbc29dc03e.png)

By commenting out that pattern all `#` form a single token and the test file parses in 0,9ms instead of 1,2ms. It's not enough for a fair benchmark, I know.

![grafik](https://user-images.githubusercontent.com/16542113/132948316-474788bc-a727-4350-a712-dc9d0213b387.png)

Reply:
makes sense. Should we perhaps remove the `meta.literal` altogether then? If it is needed by a plugin, anything unscoped is a literal ;)

Reply:
A pattern is normally scoped `string.regexp`, `meta.string.regexp string.quoted` or `string.quoted source.regexp`, which is some kind of literal with all regexp specific operators stacked on top. 

Hence it's not crucial.

I'd vote to remove it.